### PR TITLE
API docs updates

### PR DESCRIPTION
**Pull request by chrislovecnm**
- tweak to .gitignore
- updating api docs
- updated docs/apireference/README.md with how to generate docs

@justinsb any idea what happened to the hack script? `hack/make-gendocs.sh` did not work last time I tried to do this, but I updated the instructions in the `docs/apireference/README.md` for how to get it to work now.

### CODE REVIEW

**Comment by chrislovecnm** [comment]
@pwittrock is there an easy way to host these docs files? Any instructions?

**Comment by chrislovecnm** [comment]
Marked as blocks next, as we need to release updated api docs with every release.

**Comment by justinsb** [comment]
@chrislovecnm uploaded the missing hack script - sorry.  #3567

**Comment by justinsb** [comment]
LGTM but there's a conflict on .gitignore (because of the bazel ignores - sorry!), and let's call the hack script once we merge #3567 :-)

File: Makefile [hunk 1]
```diff
@@ -309,9 +309,9 @@ gen-cli-docs: ${KOPS} # Regenerate CLI docs
 	${KOPS} genhelpdocs --out docs/cli
 
 .PHONY: gen-api-docs
-gen-api-docs:
+gen-api-docs: 
 	# Follow procedure in docs/apireference/README.md
-	hack/make-gendocs.sh
+	# hack/make-gendocs.sh
```
Comment (justinsb) [inline Makefile]: We do want this.  Sorry I left it off before.

File: docs/apireference/README.md [hunk 2]
```diff
@@ -13,8 +13,11 @@ go get -u github.com/kubernetes-incubator/apiserver-builder/cmd/...
 # Install the reference docs commands (apiserver-builder commands invoke these)
 go get -u github.com/kubernetes-incubator/reference-docs/gen-apidocs/...
```
Comment (justinsb) [inline docs/apireference/README.md]: We should also mention here that we expect `make gen-api-docs` to work, presuming we think it now works!

**Comment by chrislovecnm** [comment]
@justinsb before we merge, should we make this as part of the release process??

File: docs/apireference/README.md (hunk 2 above)
Comment (chrislovecnm) [reply docs/apireference/README.md]: Another PR?  We need to figure out the install instructions.  I installed this differently than you did ...

I had to install `lister-gen`. @pwittrock any ideas?


```
go install k8s.io/code-generator/cmd/lister-gen
```

**Comment by justinsb** [comment]
@chrislovecnm putting this into the release doc is a great idea, as I think that's the right time to publish these.  I don't necessarily want to be regenerating these on every PR, depending on how stable the output is...

Looking at the PR itself now!

**Pull request merged**